src/pybind/mgr/cephadm/agent.py

```diff
@@ -719,12 +719,34 @@ def stop(self) -> None:
         self.unsubscribe()
         super().stop()
 
+    def get_data(self) -> Optional[dict]:
+        import io
+        import gzip
+        try:
+            content_encoding = cherrypy.request.headers.get('Content-Encoding', '')
+            remote_ip = cherrypy.request.remote.ip
+            raw_body = cherrypy.request.body.read()
+            if content_encoding == 'gzip':
+                self.mgr.log.info(f">>> Received gzipped payload from {remote_ip}")
+                buf = io.BytesIO(raw_body)
+                with gzip.GzipFile(fileobj=buf) as gz:
```

Comment:
That said, I'd rather go with `zstd`, since it achieves same compression as gzip by 10 times faster/less CPU. 

`zstd` has been recently (v3.14) added to the [std Python library](https://docs.python.org/es/dev/library/compression.zstd.html), but for older versions, you'd have to use the `python3-std` one.

Reply:
@epuertat as we agreed in the discussion on slack let's go in this first MVP with gzip, then we will be adding zstd support later.